### ISSUE TITLE
Is it a better solution to acess one of the spines by class atrribute?

### ISSUE THREAD

**Issue by yech1990** [matplotlib/matplotlib#17100]
hiding spines is very common in creating a new plot by matplotlib, and the flowing lines of code are widely used.

```python
ax.spines['top'].set_visible(False)
ax.spines['right'].set_visible(False)
```

They are a little comprehensive for new users to remember, and  is easy to get wrong.

Will it possible to access the spine by class attribute and make matplotlib more object-oriented, with this mirror change.

```python
ax.spines.top.set_visible(False)
ax.spines.right.set_visible(False)
```

or even,

```python
ax.spines.top = None
ax.spines.right = None
```

**Comment by nmay231** [comment]
Additionally, it would be convenient to modify all spines with a catchall attribute.

```python
ax.spines.all.set_visible(False)
```

(Though, this might need to be moved to its own issue)

**Comment by timhoffm** [comment]
`ax.spines.all.set_visible(False)` is addressed in #17107.

**Comment by nmay231** [comment]
Neat! Thanks @timhoffm :+1:

**Comment by QuLogic** [comment]
Seems like this was implemented by #17107.
